src/data/visualizations/VisualizationsD2Repository.ts

```diff
@@ -0,0 +1,37 @@
+import { Future } from "../../domain/common/entities/Future";
+import { Id } from "../../domain/common/entities/Schemas";
+import { Instance } from "../../domain/instance/entities/Instance";
+import { Visualization } from "../../domain/visualizations/entities/Visualization";
+import {
+    GetVisualizationsOptions,
+    VisualizationsRepository,
+} from "../../domain/visualizations/repositories/VisualizationsRepository";
+import { D2Api } from "../../types/d2-api";
+import { getD2APiFromInstance } from "../../utils/d2-utils";
+import { apiToFuture } from "../common/utils/api-futures";
+
+export class VisualizationsD2Repository implements VisualizationsRepository {
```

Comment:
[minor] Jorge recommends using singular name for repository: "VisualizationRepository". But check the rest of the project and follow the same idea.